libs/langchain/src/agents/middlewareAgent/middleware/hitl.ts

```diff
@@ -1,33 +1,118 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { z } from "zod/v3";
-import { v4 as uuid } from "uuid";
 import { AIMessage, ToolMessage } from "@langchain/core/messages";
 import { interrupt } from "@langchain/langgraph";
 
 import { createMiddleware } from "../middleware.js";
-import type { ToolCall } from "../types.js";
+
+const ToolConfigSchema = z.object({
+  allowAccept: z.boolean().optional(),
+  allowEdit: z.boolean().optional(),
+  allowRespond: z.boolean().optional(),
+  description: z.string().optional(),
+});
+
+type ToolConfigSchema = z.input<typeof ToolConfigSchema>;
+type ToolCall = NonNullable<AIMessage["tool_calls"]>[number];
 
 /**
- * Interrupt request for tool approval
+ * Configuration that defines what actions are allowed for a human interrupt.
+ * This controls the available interaction options when the graph is paused for human input.
  */
-export interface ToolApprovalRequest {
+export interface HumanInTheLoopConfig
+  extends Omit<ToolConfigSchema, "description"> {}
+
+/**
+ * Represents a request with a name and arguments.
+ */
+export interface ActionRequest {
+  /**
+   * The type or name of action being requested (e.g., "add_numbers")
+   */
   action: string;
+  /**
+   * Key-value pairs of arguments needed for the action (e.g., {"a": 1, "b": 2})
+   */
   args: Record<string, any>;
-  toolCallId: string;
+}
+
+/**
+ * Represents an interrupt triggered by the graph that requires human intervention.
+ *
+ * @param actionRequest - The specific action being requested from the human
+ * @param config - Configuration defining what response types are allowed
+ * @param description - Optional detailed description of what input is needed
```

Comment:
We could check what agent inbox does, but I suspect that the description isn't for what input is needed, but generally to describe the result of performing the action. i.e., describe something non obvious about the action (e.g., send_email, after approval the email will be scheduled for sending but may not be sent for up to 5 mins)

---

I am not sure about this part, I think it depends a bit how the UI uses it, but this what I'd use the description for personally